crates/starknet-devnet/tests/test_account_impersonation.rs

```diff
@@ -0,0 +1,315 @@
+
+        let casm_path = format!("{}/{}", env!("CARGO_MANIFEST_DIR"), CAIRO_1_CASM_PATH);
+        let casm_contract_definition: CompiledClass =
+            serde_json::from_reader(std::fs::File::open(casm_path).unwrap()).unwrap();
+        let compiled_class_hash = (casm_contract_definition.class_hash()).unwrap();
+        assert_eq!(Felt::from(compiled_class_hash).to_prefixed_hex_str(), CASM_COMPILED_CLASS_HASH);
+
+        account.set_block_id(BlockId::Tag(BlockTag::Latest));
+
+        // We need to flatten the ABI into a string first
+        let flattened_class = contract_artifact.flatten().unwrap();
+
+        let declare_result =
+            account.declare(Arc::new(flattened_class), compiled_class_hash).send().await;
+
+        match expected_result {
+            TestCaseResult::Success => {
+                crate::common::utils::assert_tx_successful(
+                    &declare_result.unwrap().transaction_hash,
+                    &forked_devnet.json_rpc_client,
+                )
+                .await;
+            }
+            TestCaseResult::Failure { msg } => {
+                let err = declare_result.err().unwrap();
+                assert!(format!("{:?}", err).to_lowercase().contains(&msg));
+            }
+        }
+    }
+
+    async fn test_invoke_transaction(
+        origin_devnet: &BackgroundDevnet,
+        impersonation_actions: &[ImpersonationAction],
+        expected_result: TestCaseResult,
+    ) {
+        let forked_devnet = origin_devnet.fork().await.unwrap();
+
+        let account =
+            get_account_interacting_with_forked_devnet(origin_devnet, &forked_devnet).await;
+
+        for action in impersonation_actions.iter() {
+            forked_devnet.impersonate_account(action).await.unwrap();
+        }
+
+        let forked_account_initial_balance =
+            forked_devnet.get_balance(&account.address(), FeeUnit::FRI).await.unwrap();
+
+        let invoke_call = get_invoke_transaction_request(AMOUNT_TO_TRANSFER);
+
+        let result = account.execute(vec![invoke_call]).send().await;
+
+        match expected_result {
+            TestCaseResult::Success => {
+                let result = result.unwrap();
+
+                let receipt = forked_devnet
+                    .json_rpc_client
+                    .get_transaction_receipt(result.transaction_hash)
+                    .await
+                    .unwrap();
+
+                assert_eq!(receipt.execution_result(), &ExecutionResult::Succeeded);
+                let forked_account_balance =
+                    forked_devnet.get_balance(&account.address(), FeeUnit::FRI).await.unwrap();
+                assert!(
+                    forked_account_initial_balance >= AMOUNT_TO_TRANSFER + forked_account_balance
+                );
+            }
+            TestCaseResult::Failure { msg } => {
+                let err = result.err().unwrap();
+                assert!(format!("{:?}", err).to_lowercase().contains(&msg));
```

Comment:
I see that here `to_lowercase()` is used, but could we reuse the existing utility function `fn assert_contains`?

Reply:
[6c62857](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/436/commits/6c62857000e21a4dd72873c1b2c8f6510ee0f11a)